editor/plugins/skeleton_3d_editor_plugin.cpp

```diff
@@ -666,22 +804,255 @@ void Skeleton3DEditor::_bind_methods() {
+				} else {
+					skeleton->set_subgizmo_selection(seg, 0);
+				}
+				break;
+			}
+		}
+
+	} else {
+		_hide_handles();
+		skeleton->clear_subgizmo_selection();
+	}
+}
+
+void Skeleton3DEditor::rest_mode_toggled(const bool pressed) {
+	bool before_val = rest_mode;
+
+	// Prevent that bone pose will be undo during rest mode.
+	// However SkeletonEditor will be memdeleted,
+	// so it need to record in SpatialEditor with calling method in
+	// EditorInspectorPluginSkeleton and it will not be memdeleted.
+	UndoRedo *ur = Node3DEditor::get_singleton()->get_undo_redo();
+	ur->create_action(TTR("Toggled Rest Mode"));
+	set_rest_mode_toggled(pressed);
+	ur->add_undo_method(editor_plugin, "set_rest_mode_toggled", before_val);
+	ur->add_do_method(editor_plugin, "set_rest_mode_toggled", pressed);
+	ur->commit_action();
+}
+
+void Skeleton3DEditor::set_rest_mode_toggled(const bool pressed, const bool destructing) {
+	rest_mode_button->disconnect("toggled", callable_mp(this, &Skeleton3DEditor::rest_mode_toggled));
+	rest_mode_button->set_pressed(pressed);
+	rest_mode_button->connect("toggled", callable_mp(this, &Skeleton3DEditor::rest_mode_toggled));
+
+	rest_mode = pressed;
+	const int bone_len = skeleton->get_bone_count();
+	for (int i = 0; i < bone_len; i++) {
+		skeleton->set_bone_enabled(i, !rest_mode);
+	}
+	if (!destructing) {
+		if (pose_editor) {
+			pose_editor->set_read_only(rest_mode);
+		}
+		if (custom_pose_editor) {
+			custom_pose_editor->set_read_only(rest_mode);
+		}
+	}
+
+	set_keyable(AnimationPlayerEditor::get_singleton()->get_track_editor()->has_keying() && !rest_mode);
 }
 
 Skeleton3DEditor::Skeleton3DEditor(EditorInspectorPluginSkeleton *e_plugin, EditorNode *p_editor, Skeleton3D *p_skeleton) :
 		editor(p_editor),
 		editor_plugin(e_plugin),
 		skeleton(p_skeleton) {
+	singleton = this;
+
+	handle_material = Ref<ShaderMaterial>(memnew(ShaderMaterial));
+	handle_shader = Ref<Shader>(memnew(Shader));
+	handle_shader->set_code(" \
+		shader_type spatial; \
+		render_mode unshaded, shadows_disabled, depth_draw_always; \
+		uniform vec4 albedo : hint_color = vec4(1,1,1,1); \
+		uniform sampler2D texture_albedo : hint_albedo; \
+		uniform float point_size : hint_range(0,128) = 32; \
+		void vertex() { \
+			if (!OUTPUT_IS_SRGB) { \
+				COLOR.rgb = mix( pow((COLOR.rgb + vec3(0.055)) * (1.0 / (1.0 + 0.055)), vec3(2.4)), COLOR.rgb* (1.0 / 12.92), lessThan(COLOR.rgb,vec3(0.04045)) ); \
+			} \
+			VERTEX = VERTEX; \
+			POSITION=PROJECTION_MATRIX*INV_CAMERA_MATRIX*WORLD_MATRIX*vec4(VERTEX.xyz,1.0); \
+			POSITION.z = mix(POSITION.z, 0, 0.999); \
+			POINT_SIZE = point_size; \
+		} \
+		void fragment() { \
+			vec4 albedo_tex = texture(texture_albedo,POINT_COORD); \
+			vec3 col = albedo_tex.rgb + COLOR.rgb; \
+			col = vec3(min(col.r,1.0),min(col.g,1.0),min(col.b,1.0)); \
+			ALBEDO = albedo.rgb * col; \
+			if (albedo.a * albedo_tex.a < 0.5) { discard; } \
+			ALPHA = albedo.a * albedo_tex.a; \
+		} \
```

Comment:
The `albedo` uniform is unused, it should be removed. The rest looks great. 

There are plans to add an easier way to do this kind of Z-scaling, but until we add it, this is the only way to achieve the wanted effect.